File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
    int[] ans = new int[nums.length];
    int prod = 1;
    List<Integer> id = new ArrayList<>();
    for (int i = 0; i < nums.length; i++) {
        if (nums[i] == 0) {
            id.add(i);
            continue;
        }
        prod *= nums[i];
    }
    if (id.size() == 1)
        ans[id.get(0)] = prod;
    else if (id.size() == 0) {
        for (int i = 0; i < nums.length; i++) {
            ans[i] = prod / nums[i];
        }
    }
    return ans;
}
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans = [0] * len(nums)
        prod = 1
        id = []
        for i in range(len(nums)):
            if nums[i] == 0:
                id.append(i)
                continue
            prod *= nums[i]
        if len(id) == 1:
            ans[id[0]] = prod
        elif len(id) == 0:
            for i in range(len(nums)):
                ans[i] = prod // nums[i]
        return ans